common/src/primitives/amount.rs

```diff
@@ -226,6 +231,12 @@ impl std::ops::Shr<u32> for Amount {
     }
 }
 
+impl Sum for Amount {
+    fn sum<I: Iterator<Item = Self>>(iter: I) -> Self {
+        iter.fold(Amount::zero(), |a, b| (a + b).expect("Overflow operation"))
```

Comment:
I think it's better to throw an error here than `expect`.
For example:
```
impl std::iter::Sum<Amount> for Option<Amount> {
    fn sum<I>(mut iter: I) -> Self
    where
        I: Iterator<Item = Amount>,
    {
        iter.try_fold(Amount::new(0), std::ops::Add::add)
    }
}
```

Reply:
Correct. There seems to be a misunderstanding on when to use expect/unwrap. We should use them only if our code model deems the situation impossible. There's no reason to think an overflow is impossible when adding a list of iterable items.